core/src/main/java/org/apache/iceberg/encryption/EncryptionUtil.java

```diff
@@ -70,31 +76,105 @@ public static KeyManagementClient createKmsClient(Map<String, String> catalogPro
+
+    if (null == tableKeyId) {
+      // Unencrypted table
+      return PlaintextEncryptionManager.instance();
+    }
+
     Preconditions.checkState(
         dataKeyLength == 16 || dataKeyLength == 24 || dataKeyLength == 32,
         "Invalid data key length: %s (must be 16, 24, or 32)",
         dataKeyLength);
 
-    return new StandardEncryptionManager(tableKeyId, dataKeyLength, kmsClient);
+    return new StandardEncryptionManager(tableKeyId, dataKeyLength, kmsClient, kekCacheTimeout);
   }
 
   public static EncryptedOutputFile plainAsEncryptedOutput(OutputFile encryptingOutputFile) {
     return new BaseEncryptedOutputFile(encryptingOutputFile, EncryptionKeyMetadata.empty());
   }
+
+  public static EncryptionKeyMetadata setFileLength(
+      EncryptionKeyMetadata encryptionKeyMetadata, long manifestListLength) {
+    Preconditions.checkState(
+        encryptionKeyMetadata instanceof StandardKeyMetadata,
+        "Cant set file length in %s",
+        encryptionKeyMetadata.getClass());
+    ((StandardKeyMetadata) encryptionKeyMetadata).setFileLength(manifestListLength);
+    return encryptionKeyMetadata;
+  }
+
+  private static long parseFileLength(ByteBuffer keyMetadataBuffer) {
+    StandardKeyMetadata standardKeyMetadata = StandardKeyMetadata.parse(keyMetadataBuffer);
+    return standardKeyMetadata.fileLength();
+  }
+
+  public static void getKekCacheFromMetadata(FileIO io, Map<String, KeyEncryptionKey> kekCache) {
+    Preconditions.checkState(
+        io instanceof EncryptingFileIO,
+        "Can't set KEK cache - IO %s is not instance of EncryptingFileIO",
+        io.getClass());
+    EncryptionManager encryption = ((EncryptingFileIO) io).encryptionManager();
+    Preconditions.checkState(
+        encryption instanceof StandardEncryptionManager,
+        "Can't set KEK cache - encryption manager %s is not instance of StandardEncryptionManager",
+        encryption.getClass());
+    ((StandardEncryptionManager) encryption).addKekCache(kekCache);
+  }
+
+  public static InputFile decryptManifestListFile(
+      ManifestListFile manifestListFile, FileIO fileIO) {
+    Preconditions.checkArgument(
+        fileIO instanceof EncryptingFileIO,
+        "Cannot read manifest list (%s) because it is encrypted but the configured "
+            + "FileIO (%s) does not implement EncryptingFileIO",
+        manifestListFile.location(),
+        fileIO.getClass());
+    EncryptingFileIO encryptingFileIO = (EncryptingFileIO) fileIO;
+
+    Preconditions.checkArgument(
+        encryptingFileIO.encryptionManager() instanceof StandardEncryptionManager,
+        "Cannot read manifest list (%s) because it is encrypted but the "
+            + "encryption manager (%s) is not StandardEncryptionManager",
+        manifestListFile.location(),
+        encryptingFileIO.encryptionManager().getClass());
+    StandardEncryptionManager standardEncryptionManager =
+        (StandardEncryptionManager) encryptingFileIO.encryptionManager();
```

Comment:
Similar to the comment just above, if this is done in `EncryptingFileIO` then there is no need to cast.

Reply:
EncryptingFileIO is in the api module. Do we want to move it to the core module, or create an extending class there? So we'll have access to the functionality required for manifest list encryption (KEKs/cache, and file length in key metadata).